2008-09-13  Doug Goldstein  <cardoe@gentoo.org>
	* switch to udevadm settle since udevsettle has been deprecated
	  and finally removed with udev 126.

--- cryptsetup-1.0.6/lib/libdevmapper.c
+++ cryptsetup-1.0.6/lib/libdevmapper.c
@@ -18,11 +18,11 @@
 
 #define	CRYPT_TARGET	"crypt"
 
-#define UDEVSETTLE	"/sbin/udevsettle"
+#define UDEVSETTLE	"/sbin/udevadm settle"
 
-static void run_udevsettle(void)
+static int run_udevsettle(void)
 {
-	system(UDEVSETTLE);
+	return system(UDEVSETTLE);
 }
 
 static void set_dm_error(int level, const char *file, int line,
